src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -59,9 +59,11 @@
 register_cli_argument('resource group', 'resource_group_name', resource_group_name_type, options_list=('--name', '-n'))
 register_cli_argument('resource group deployment', 'resource_group_name', arg_type=resource_group_name_type, completer=get_resource_group_completion_list)
 register_cli_argument('resource group deployment', 'deployment_name', options_list=('--name', '-n'), required=True, help='The deployment name.')
-register_cli_argument('resource group deployment', 'parameters_file_path', completer=FilesCompleter())
-register_cli_argument('resource group deployment', 'template_file_path', completer=FilesCompleter())
+register_cli_argument('resource group deployment', 'parameters_file_path', completer=FilesCompleter(), help='provide deployment parameter values, a file path, or an uri to a remote file, or json string')
```

Comment:
This should work with the @ syntax, which should result in simplified logic. @ will handle loading your file as a JSON string, so you don't need to do that manually anymore. You will probably still need to check if it is a URI (though it might work with @?)

Also, this help text implies to me that there are 4 ways of populating this argument:
1. provide parameter values (key value pairs?)
2. provide a file path (recommend just using @ syntax)
3. provide a remote uri (try with @ syntax)
4. provide JSON string (the one no one would ever do, but because of @ syntax should be the default).

If @ syntax doesn't work with uri's as is, it might make sense to extend it to do so.